src/sentry/utils/snuba.py

```diff
@@ -764,6 +790,100 @@ def _snuba_query(params):
         raise SnubaError(err)
 
 
+def _snql_query(params: Tuple[SnubaQuery, Hub, Mapping[str, str]]) -> RawResult:
+    # Run the SnQL query in debug/dry_run mode
+    # Run the legacy query in debug mode
+    # Log any errors in SnQL execution and log if the returned SQL is not the same
+    query_data, thread_hub, headers = params
+    query_params, forward, reverse = query_data
+    og_debug = query_params.get("debug", False)
+    try:
+        query = json_to_snql(query_params, query_params["dataset"])
+        query.validate()  # Call this here just avoid it happening in the async all
+    except Exception as e:
+        logger.warning(
+            "snuba.snql.parsing.error",
+            extra={"error": str(e), "params": json.dumps(query_params)},
+        )
+        return _snuba_query(params)
+
+    query = query.set_dry_run(True).set_debug(True)
+    query_params["debug"] = True
+
+    legacy_future = _query_thread_pool.submit(_snuba_query, (query_data, Hub(thread_hub), headers))
+    snql_future = _query_thread_pool.submit(_raw_snql_query, query, Hub(thread_hub), headers)
```

Comment:
Does it make sense at all to just submit `snql_future` to the thread pool then call `_snuba_query` directly?

Reply:
Yeah that's a little cleaner I think.